<launch>

  <!-- args corresponding to environment variables -->
  <arg name="PROFILER" default="$(optenv PROFILER false)" />
  <arg name="UAV_NAME" default="$(optenv UAV_NAME)" />
  <arg name="RUN_TYPE" default="$(optenv RUN_TYPE)" />
  <arg name="UAV_TYPE" default="$(optenv UAV_TYPE)" />
  <arg name="LOGGER_DEBUG" default="$(optenv LOGGER_DEBUG false)" />

    <!-- other args -->
  <arg name="custom_config" default="" />
  <arg name="standalone" default="true" />
  <arg name="debug" default="false" />

  <env name="ROSCONSOLE_CONFIG_FILE" if="$(eval arg('LOGGER_DEBUG'))" value="$(find mrs_uav_managers)/config/debug_verbosity.yaml" />

  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="load" />
  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="$(arg UAV_NAME)_nodelet_manager" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <group ns="$(arg UAV_NAME)">

    <!-- Gain Manager -->
    <node pkg="nodelet" type="nodelet" name="gain_manager" args="$(arg nodelet) mrs_uav_managers/GainManager $(arg nodelet_manager)" output="screen" launch-prefix="$(arg launch_prefix)">

      <!-- Load the particular param files -->
      <rosparam file="$(find mrs_uav_managers)/config/default/gain_manager/gain_manager.yaml" />
      <rosparam file="$(find mrs_uav_managers)/config/default/gain_manager/gains.yaml" />

        <!-- Load the particular param files -->
      <rosparam file="$(find mrs_uav_managers)/config/$(arg RUN_TYPE)/$(arg UAV_TYPE)/gain_manager/gain_manager.yaml" />
      <rosparam file="$(find mrs_uav_managers)/config/$(arg RUN_TYPE)/$(arg UAV_TYPE)/gain_manager/gains.yaml" />

      <!-- Load a user param file -->
      <rosparam if="$(eval not arg('custom_config') == '')" file="$(arg custom_config)" />

      <param name="enable_profiler" type="bool" value="$(arg PROFILER)" />

      <!-- Subscribers -->
      <remap from="~control_manager_diagnostics_in" to="control_manager/diagnostics" />
      <remap from="~odometry_diagnostics_in" to="estimation_manager/diagnostics_legacy" />

      <!-- Services -->
      <remap from="~set_gains_in" to="~set_gains" />
      <remap from="~set_gains_out" to="control_manager/se3_controller/set_parameters" />

      <!-- Publishers -->
      <remap from="~diagnostics_out" to="~diagnostics" />
      <remap from="~profiler" to="profiler" />

    </node>

  </group>

</launch>
